src/sentry/static/sentry/app/views/events/utils/getParams.tsx

```diff
@@ -1,20 +1,84 @@
 import {DEFAULT_STATS_PERIOD} from 'app/constants';
 import {defined} from 'app/utils';
+import moment from 'moment';
 
-const getUtcValue = utc => {
+const STATS_PERIOD_PATTERN = '^\\d+[hdmsw]?$';
+
+function validStatsPeriod(input: string) {
+  return !!input.match(STATS_PERIOD_PATTERN);
+}
+
+const getStatsPeriodValue = (
+  maybe: string | string[] | undefined | null
+): string | undefined => {
+  if (Array.isArray(maybe)) {
+    if (maybe.length <= 0) {
+      return undefined;
+    }
+
+    return maybe.find(validStatsPeriod);
+  }
+
+  if (typeof maybe === 'string' && validStatsPeriod(maybe)) {
+    return maybe;
+  }
+
+  return undefined;
+};
+
+const getDateTimeString = (
+  maybe: string | string[] | undefined | null
+): string | undefined => {
```

Comment:
This and a few of the other parsers in this file have similar function bodies with only the validator/parsing operation being different. Could we extract that duplication?